docs/docs/how-tos/monitoring.md

```diff
@@ -0,0 +1,129 @@
+
+![Grafana Select Loki](/img/how-tos/2_grafana-select-loki.png)
+
+Click on the Log browser and select labels to search in, in this case we have selected pod:
+
+![Grafana Select Loki](/img/how-tos/3_grafana-log-browser-pod.png)
+
+Type the pod name initials for the pod you're looking to search logs for, in this case we are looking
+for hub pod:
+
+![Grafana Select Loki](/img/how-tos/4_grafana-log-search-pod.png)
+
+Select the pod from the list of pods and then click on "Show logs":
+
+![Grafana Select Loki](/img/how-tos/5_grafana-log-select-pod.png)
+
+After clicking on "Show logs", you should be able to see logs for JupyterHub pod as shown below:
+
+![Grafana Select Loki](/img/how-tos/6_grafana-view-pod-logs.png)
+
+You can also filter by time, by clicking on the time filter on top right, next to "Run query".
+
+## Terraform Overrides
+
+Nebari provides its users with the ability to customize the deployment of various component
+and Loki is one of them. Loki deployment is made up of three fundamental components:
+
+- Loki: a set of components that when composed forms a fully featured logging stack
+- Promtail: an agent which ships the contents of local logs to a Loki instance
+- MinIO: a Kubernetes-native high-performance object storage server which is designed for large-scale
+  private cloud infrastructure and compatible with Amazon S3.
+
+```yaml
+monitoring:
+  enabled: true
+  overrides:
+    loki: <LOKI-HELM-CHART-VALUES-OVERRIDE>
+    promtail: <PROMTAIL-HELM-CHART-VALUES-OVERRIDE>
+    minio: <MINIO-HELM-CHART-VALUES-OVERRIDE>
+```
+
+Below are some examples of customizing your loki deployment via terraform overrides:
+
+### Log Storage
+
+You can customize the size of persistent volume for logs storage, by setting up the value for
+`minio.persistence.size` as shown below:
+
+```yaml
+monitoring:
+    enabled: true
+    overrides:
+      minio:
+        persistence:
+          size: 100Gi
+```
+
+### Log Retention Period
+
+Since the storage for MinIO is predefined and is not infinitely scalable out-of-the-box like
+cloud storage solutions like S3, GCS, etc, unless you keep increasing minio persistent volume size.
+It is a good idea to clear up old logs to make space for new ones, Nebari does this by default with
+60 days retention period. You can customize the log retention period by overriding the default
+value as follows:
+
+```yaml
+monitoring:
+    enabled: true
+    overrides:
+      loki:
+        loki:
+          limits_config:
+            # The minimum retention period is 24h.
+            retention_period: 90d
+```
+
+This may not be a suitable solution for many cases, like if your organization needs to keep
+all the logs forever, then you may want to use a very large persistent volume and disable
+deletion of old logs. You can disable deletion via:
+
+```yaml
+monitoring:
+    enabled: true
+    overrides:
+      loki:
+        loki:
+          compactor:
+            retention_enabled: false
+```
```

Comment:
Might be interesting to include a reference to loki documentation on switching the storage type from minio to another compatible S3 Object storage utility

Reply:
Done